### PR TITLE
agent-client-js: update doc with withRefs API

### PR DESCRIPTION
**Pull request by such**
This change is [Reviewable](https://reviewable.io/reviews/stratumn/indigo-js/140)

### CODE REVIEW

**Comment by t-bast** [comment]
Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion.

---

*[packages/agent-client-js/README.md, line 154 at r1](https://reviewable.io:443/reviews/stratumn/indigo-js/140#-KzZhhpRPB9CTGipWik2:-KzZhhpRPB9CTGipWik3:baqbo92) ([raw file](https://github.com/stratumn/indigo-js/blob/f2c7eb81b20031b6fc443a995c85b0f8ab72a993/packages/agent-client-js/README.md#L154)):*
> ```Markdown
>   .then(function(agent) {
>     const process = agent.processes.firstProcess;
>     process.createMap('A new map');
> ```

I'm not sure I understand the content of this then...
Since process.createMap returns a promise, here you start it but don't watch for the outcome...
Shouldn't you instead keep the existing code and add a new ".then" with the refs example in it?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/stratumn/indigo-js/140)*

**Comment by t-bast** [comment]
Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/stratumn/indigo-js/140)*

**Comment by such** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[packages/agent-client-js/README.md, line 154 at r1](https://reviewable.io:443/reviews/stratumn/indigo-js/140#-KzZhhpRPB9CTGipWik2:-KzZiOcIf_-kFIwnUOZ7:b4zjv9q) ([raw file](https://github.com/stratumn/indigo-js/blob/f2c7eb81b20031b6fc443a995c85b0f8ab72a993/packages/agent-client-js/README.md#L154)):*
<details><summary><i>Previously, t-bast (Bastien Teinturier) wrote…</i></summary><blockquote>

I'm not sure I understand the content of this then...
Since process.createMap returns a promise, here you start it but don't watch for the outcome...
Shouldn't you instead keep the existing code and add a new ".then" with the refs example in it?
</blockquote></details>

Yeah I wanted to keep things simple because if I add a new then, then I won't really use the return value of the first call to createMap. But I guess I you find that version confusing we should work on something better.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/stratumn/indigo-js/140)*

**Comment by t-bast** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[packages/agent-client-js/README.md, line 154 at r1](https://reviewable.io:443/reviews/stratumn/indigo-js/140#-KzZhhpRPB9CTGipWik2:-KzZiyWbNRXB97La89Hz:bb592eu) ([raw file](https://github.com/stratumn/indigo-js/blob/f2c7eb81b20031b6fc443a995c85b0f8ab72a993/packages/agent-client-js/README.md#L154)):*
<details><summary><i>Previously, such (Adrien Montfort) wrote…</i></summary><blockquote>

Yeah I wanted to keep things simple because if I add a new then, then I won't really use the return value of the first call to createMap. But I guess I you find that version confusing we should work on something better.
</blockquote></details>

Maybe you can simply remove the process.createMap('A new map') and only show the one with refs, and in the comment mention that the withRefs is optional ?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/stratumn/indigo-js/140)*

**Comment by conord33** [comment]
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[packages/agent-client-js/README.md, line 154 at r1](https://reviewable.io:443/reviews/stratumn/indigo-js/140#-KzZhhpRPB9CTGipWik2:-KzZl94OjYcGSOq2VkV9:bdb2j75) ([raw file](https://github.com/stratumn/indigo-js/blob/f2c7eb81b20031b6fc443a995c85b0f8ab72a993/packages/agent-client-js/README.md#L154)):*
<details><summary><i>Previously, t-bast (Bastien Teinturier) wrote…</i></summary><blockquote>

Maybe you can simply remove the process.createMap('A new map') and only show the one with refs, and in the comment mention that the withRefs is optional ?
</blockquote></details>

you could maybe just comment out the second one so its still executable code but shows an example of each?

---

*[packages/agent-client-js/README.md, line 156 at r1](https://reviewable.io:443/reviews/stratumn/indigo-js/140#-KzZkksrTPDFvxv5H-te:-KzZkksrTPDFvxv5H-tf:b-3muh46) ([raw file](https://github.com/stratumn/indigo-js/blob/f2c7eb81b20031b6fc443a995c85b0f8ab72a993/packages/agent-client-js/README.md#L156)):*
> ```Markdown
>     process.createMap('A new map');
> 
>     // you can also add references to that first segment
> ```

i would rephrase to `you can also add references to the first segment`. that makes it seem like it already exists and you are updating it

---

*[packages/agent-client-js/README.md, line 276 at r1](https://reviewable.io:443/reviews/stratumn/indigo-js/140#-KzZlI6wHrOg265fpx5w:-KzZlI6wHrOg265fpx5x:bdds9ku) ([raw file](https://github.com/stratumn/indigo-js/blob/f2c7eb81b20031b6fc443a995c85b0f8ab72a993/packages/agent-client-js/README.md#L276)):*
> ```Markdown
>   })
>   .then(function(segment) {
>     // you can also add references for that segment
> ```

i would say `you can also add references to a  new segment`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/stratumn/indigo-js/140)*

**Comment by such** [comment]
Review status: 0 of 1 files reviewed at latest revision, 3 unresolved discussions.

---

*[packages/agent-client-js/README.md, line 154 at r1](https://reviewable.io:443/reviews/stratumn/indigo-js/140#-KzZhhpRPB9CTGipWik2:-KzZmjWvpuM3BY8g9s8M:bbeq5zw) ([raw file](https://github.com/stratumn/indigo-js/blob/f2c7eb81b20031b6fc443a995c85b0f8ab72a993/packages/agent-client-js/README.md#L154)):*
<details><summary><i>Previously, conord33 (Conor Dawson) wrote…</i></summary><blockquote>

you could maybe just comment out the second one so its still executable code but shows an example of each?
</blockquote></details>

I like that.
Done.

---

*[packages/agent-client-js/README.md, line 156 at r1](https://reviewable.io:443/reviews/stratumn/indigo-js/140#-KzZkksrTPDFvxv5H-te:-KzZmljpl2yfxYXnNh2f:bf0x5k0) ([raw file](https://github.com/stratumn/indigo-js/blob/f2c7eb81b20031b6fc443a995c85b0f8ab72a993/packages/agent-client-js/README.md#L156)):*
<details><summary><i>Previously, conord33 (Conor Dawson) wrote…</i></summary><blockquote>

i would rephrase to `you can also add references to the first segment`. that makes it seem like it already exists and you are updating it
</blockquote></details>

I went the past perfect way... Is this correct?

---

*[packages/agent-client-js/README.md, line 276 at r1](https://reviewable.io:443/reviews/stratumn/indigo-js/140#-KzZlI6wHrOg265fpx5w:-KzZmnJ1cOnhQ4v-Uzyd:b-896fix) ([raw file](https://github.com/stratumn/indigo-js/blob/f2c7eb81b20031b6fc443a995c85b0f8ab72a993/packages/agent-client-js/README.md#L276)):*
<details><summary><i>Previously, conord33 (Conor Dawson) wrote…</i></summary><blockquote>

i would say `you can also add references to a  new segment`
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/stratumn/indigo-js/140)*

**Comment by conord33** [comment]
:lgtm:

---

Review status: 0 of 1 files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/stratumn/indigo-js/140#-:-KzZn1apDdsf-pNirP7W:bnfp4nl)*

**Comment by t-bast** [comment]
Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/stratumn/indigo-js/140)*
